roles/rsyslog/tasks/main.yml

```diff
@@ -1,3 +1,7 @@
+- name: Include top vars
+  include_vars:
+    file: "{{ role_path | dirname | dirname }}/defaults/main.yml"
```

Comment:
Did we discuss this before?  That using `include_vars` makes the vars defined in the file a different precedence than `defaults`?  https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#ansible-variable-precedence - role defaults have a much lower precedence than anything else - not sure if it makes a difference here.  See also https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html#role-default-variables

The important thing is that the user can, with an inventory, or with `-e var` or `-e@vars.yml`, override any of the vars defined in `defaults/main.yml`.  But the "Using Variables" page says this:
"If you are writing a role and want to ensure the value in the role is absolutely used in that role, and is not going to be overridden by inventory, you should put it in roles/x/vars/main.yml like so, and inventory values cannot override it. -e however, still will"

I guess the problem is that the variables defined in `defaults/vars.yml` are not available in all contexts in the child roles?  For example, with the current code, if I don't include the vars, and remove the `| d('rsyslog')` below, I get this:
```
RUNNING HANDLER [/home/rmeggins/linux-system-roles/logging/tests/roles/linux-system-roles.logging/roles/rsyslog : restart rsyslogd] ***
task path: /home/rmeggins/linux-system-roles/logging/roles/rsyslog/handlers/main.yml:2
fatal: [/var/lib/libvirt/images/f31.qcow2]: FAILED! => {}

MSG:

The conditional check 'logging_collector == 'rsyslog'' failed. The error was: error while evaluating conditional (logging_collector == 'rsyslog'): 'logging_collector' is undefined
```

Reply:
I did a google search for `ansible role default variables defined in parent role are not defined in subroles`
This is what I found: https://www.samdarwin.com/ansible-default-vars/ - looks very hacky, but seems to work . . .

@sradco - ovirt uses nested roles quite heavily - but even with this logging role that is much simpler than the whole of ovirt ansible, we have run into several issues - has ovirt run into issues like these with the use of nested roles?  I'm wondering why we keep running into issues like this, and if we should just assume ansible doesn't like nested roles, and rewrite the logging role not to use them . . .

Reply:
@richm, not sure if I understood the hack correctly, but with this [commit](2790ba189840d890fa984f4fd6e0f46cb061eb9f), I could verify that logging_provider could be overriden if it's set in the command line with -e as well as in the inventory.

Reply:
> @richm, not sure if I understood the hack correctly, but with this [commit](2790ba189840d890fa984f4fd6e0f46cb061eb9f), I could verify that logging_provider could be overriden if it's set in the command line with -e as well as in the inventory.

Ok - if it is working as-is, then let's go with it.

Reply:
ah, I see - never mind - are you able to override the defaults with inventory with commit https://github.com/linux-system-roles/logging/pull/110/commits/15d37bbf9560e6c59b8e3bd1832e9f3c4089b3f3 ?

Reply:
> ah, I see - never mind - are you able to override the defaults with inventory with commit [15d37bb](https://github.com/linux-system-roles/logging/commit/15d37bbf9560e6c59b8e3bd1832e9f3c4089b3f3) ?

@richm, Yes, I was.  But rethinking about this, probably, we should be able to convert the rsyslog role into a task/var pair (e.g., logging/{tasks,vars}/rsyslog/main.yml) as we did for the call-back?  If we could, it'd reduce the future potential headaches...  Let me play with it a bit.

Reply:
Ok - but I wouldn't worry about it too much, as long as with https://github.com/linux-system-roles/logging/commit/15d37bbf9560e6c59b8e3bd1832e9f3c4089b3f3 you can override the defaults with settings from your inventory.

Reply:
All right. Then, I won't spend too much time on it :). Thanks!

Reply:
I tried to converting the include_role rsyslog to include_tasks/vars, but the attempt failed since I could not figure out how to include the defaults file. If I put the default variables in the included vars file, the variable could not be overridden. :(

So, I have to give up the idea and live with this PR...

Reply:
> I tried to converting the include_role rsyslog to include_tasks/vars, but the attempt failed since I could not figure out how to include the defaults file. If I put the default variables in the included vars file, the variable could not be overridden. :(
> 
> So, I have to give up the idea and live with this PR...

ok, I think that's fine for now

Reply:
@richm, @pcahyna, can I consider the review is done for this PR? Thanks!